### DISCUSSION TITLE
How to apply preprocessing on nested DataFrames using Scikit-Learn

### DISCUSSION THREAD

**Discussion by superuser28** [alan-turing-institute/sktime#2201] [General]
Hello all,

I am not sure if my question belongs here, but I didn't know where else to ask it either.

I want to train some TSC algorithms like ROCKET and 1NN-DTW on my dataset. After converting it into the required nested DataFrame structure I wondered how I would go about standardizing/ normalizing the data prior to applying any of these TSC algorithms. I would like to include the Scikit-Learn StandardScaler as a preprocessing step in a pipeline and then run (nested) cross-validation. Is there a way to do this without having to implement a custom Scikit-Learn transformer? What is the recommended approach here?

I would appreciate any help or correction on this matter.

**Comment by fkiraly** [comment]
You can use the `TabularToSeriesAdaptor` for that afaik, and `sklearn` `Pipeline` should be compatible with both.

In the new dev version and from 0.11.0 (but not in the current 0.10.1 on pypi), you should also be able to use the `*` dunder method to compose transformers and classifiers, e.g., `pipe = TabularToSeriesAdaptor(StandardScaler()) * MyFavouriteTSC()`.

I also recognize that it should perhaps be easier to apply transformers from sklearn directly. 
For instance, `StandardScaler() * MyFavouriteTSC()` should also work. Let me see what I can cook up.

**Reply by fkiraly** [reply]
PS: this will scale each variable of each time series to mean zero and std one. I that what you wanted? Or, did you want to scale time series of equal length, so each sample composed of values at the same time index has mean zero and std one?

**Reply by superuser28** [reply]
Hey fkiraly,
yes, I believe the former is what I want to do. Thanks a lot, this will definitely help me. I also found the Tabularizer transformer earlier and just took a look into it. Transforming the data back and forth between nested and regular seemed like a possibility. Now, I will look into TabularToSeriesAdaptor as well!
Thanks again!

**Reply by fkiraly** [reply]
Yes, I think you want the `TabularToSeriesAdapter` then. The `Tabularizer` does the *latter*, not the *former*.
